etc/dovetail-edge-deploy.yml

```diff
@@ -0,0 +1,866 @@
+        - PolicyName: LambdaInfoPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - lambda:ListVersionsByFunction
+                Resource:
+                  - !Ref StagingStitchLambdaFunctionArn
+                  - !Ref StagingBytesLambdaFunctionArn
+                  - !Ref ProductionStitchLambdaFunctionArn
+                  - !Ref ProductionBytesLambdaFunctionArn
+        - PolicyName: CodePipelinePolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - codepipeline:PutJobSuccessResult
+                  - codepipeline:PutJobFailureResult
+                  - codepipeline:GetPipelineState
+                Resource: "*"
+  DiscoverEdgeLambdaVersionsFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        ZipFile: |
+          import zipfile
+          import boto3
+          import traceback
+          import json
+          import uuid
+          import os
+          from botocore.client import Config
+
+          s3 = boto3.client('s3', config=Config(signature_version='s3v4'))
+          code_pipeline = boto3.client('codepipeline')
+          lamb = boto3.client('lambda')
+
+
+          def put_job_success(job, message):
+              print('Putting job success')
+              print(message)
+              code_pipeline.put_job_success_result(jobId=job['id'])
+
+
+          def put_job_failure(job, message):
+              print('Putting job failure')
+              print(message)
+              code_pipeline.put_job_failure_result(
+                  jobId=job['id'],
+                  failureDetails={'message': message, 'type': 'JobFailed'})
+
+
+          # Queries for version information about each Dovetail edge Lambda function, and
+          # returns a dictionary with the most recent version of each.
+          def function_versions():
+              staging_stitch = lamb.list_versions_by_function(
+                  FunctionName=os.environ['STAGING_STITCH_LAMBDA_FUNCTION_ARN'],
+                  MaxItems=999
+              )
+
+              staging_bytes = lamb.list_versions_by_function(
+                  FunctionName=os.environ['STAGING_BYTES_LAMBDA_FUNCTION_ARN'],
+                  MaxItems=999
+              )
+
+              production_stitch = lamb.list_versions_by_function(
+                  FunctionName=os.environ['PRODUCTION_STITCH_LAMBDA_FUNCTION_ARN'],
+                  MaxItems=999
+              )
+
+
+              production_bytes = lamb.list_versions_by_function(
+                  FunctionName=os.environ['PRODUCTION_BYTES_LAMBDA_FUNCTION_ARN'],
+                  MaxItems=999
+              )
+
+              return {
+                'staging_bytes': staging_bytes['Versions'][-1]['Version'],
```

Comment:
So the returned order is guaranteed to be oldest to newest?

Reply:
I don't see that explicitly stated anywhere, but I've never seen it do anything else

Reply:
Might be a reason to always show changesets in slack (or at least "going from version X -> version Y") before you click the prod deploy button.  Just in case something wacky happens.

Reply:
But yeah, can tackle that later, if this works.